src/DependencyInjection/Compiler/TweakCompilerPass.php

```diff
@@ -38,7 +39,10 @@ public function process(ContainerBuilder $container): void
         /** @var string[] $defaultContexts */
         $defaultContexts = $container->getParameter('sonata_blocks.default_contexts');
 
+        /** @var array<string, Reference> $blockServiceReferences */
+        $blockServiceReferences = [];
         foreach ($container->findTaggedServiceIds('sonata.block') as $id => $tags) {
+            // NEXT_MAJOR: remove and do not make them public
             $container->getDefinition($id)
                 ->setPublic(true);
```

Comment:
I think if we remove this now it might be a BC break if someone (for whatever reason) needs them to be public? :man_shrugging: